### DISCUSSION TITLE
A very simple online editor which uses math.js

### DISCUSSION THREAD

**Discussion by felipec** [josdejong/mathjs#2636] [Show and tell]
I like the syntax of math.js, but there aren't many user-friendly options to write expressions. That's why I decided to write a very simple editor.

![screenshot](https://i.imgur.com/34m3pwz.png)

The main feature I worked on is syntax highlighting so that the text is bearable to my eyes. I also ported my vim theme to highlight.js in order to have decent colors.

You can try it out [here](https://felipec.github.io/math-notepad/), and check a basic example [here](https://felipec.github.io/math-notepad/?input=https%3A%2F%2Fraw.githubusercontent.com%2Ffelipec%2Fmath-notepad%2Fmaster%2Fexamples%2Fbasic.mathjs).

You can drag and drop files, and also specify some URL to be loaded automatically to share with someone else. For example I just uploaded an example to [dpaste.com](https://dpaste.com) [here](https://dpaste.com/GRB57ZMQZ.txt), but doesn't look very nice, nor is it very useful. Just add `?input=$URL` and you can open it in `math-noteped`: [example](https://felipec.github.io/math-notepad/?input=https://dpaste.com/GRB57ZMQZ.txt).

There's also work to make it run as a desktop application with [Electron](https://www.electronjs.org/).

The code is in [my GitHub repo](https://github.com/felipec/math-notepad), so if there's anything you don't like feel free to hack it.

Hopefully some of you will find it useful 👍

**Comment by josdejong** [comment]
Thanks for sharing Felipe!

**Comment by dvd101x** [comment]
Hi Felipe and Jos

I'm exploring the posibility of including markdown+latex to this notepad

![quadratic](https://user-images.githubusercontent.com/16261897/192566830-e770ff76-bbb3-4752-89b7-d721a63022d4.PNG)

Here you can find the fork https://github.com/dvd101x/math-notepad

The nomenclature is not ideal but it's loosely based on:
* [octave publishing markup](https://docs.octave.org/v4.2.0/Using-Publishing-Markup-in-Script-Files.html#Using-Publishing-Markup-in-Script-Files)
* [matlab publishing markup](https://la.mathworks.com/help/matlab/matlab_prog/marking-up-matlab-comments-for-publishing.html)

**Reply by josdejong** [reply]
👍 thanks for sharing David
